### PR TITLE
Paper Revisions

### PR DESCRIPTION
**Pull request by trevorb1**
This PR addresses the feedback received from reviewers in the [Global Sensitivity Analysis](https://open-research-europe.ec.europa.eu/articles/3-30) manuscript. 

The changes to be made include: 
- [x] upgrade to otoole version 1 (to improve the stability of the workflow) 
- [x] add option to scale the elementary effect, allowing for comparison between different model outputs (see PR in SALib [here](https://github.com/SALib/SALib/pull/580)) 
- [x] Add method to recreate all model runs found in the paper 
- [x] Improve running instructions
- [x] Update example notebooks to reflect the above changes 
- [ ] Revisit uncertainty range for specified annual demand in model 3

Please feel free to add to the list as well if I missed anything :)

### CODE REVIEW

**Comment by trevorb1** [comment]
Upgraded to otoole version 1.0.4! A couple notable improvements came with this
1. All default values now come from the otoole configuration files (rather than the `default_values.csv` used in older versions)
2. All input and output csvs are now directly read from the otoole configuration file. This renders the old `resources/otoole_files.csv` useless and can be deleted. 
3. Ability to easily change model files, by simply updating the otoole configuration file and not having to worry about the datapackage anymore.

**Comment by trevorb1** [comment]
I have added in a new file `workflow/rules/paper.smk` which will pull all data and files needed to recreate paper results from Zenodo and set up the repository correctly to run them. I haven't pushed the files to Zenodo yet, as we may want to discuss first, but if anybody does want to try beforehand, just let me know and I will update Zenodo. Running instructions have been updated in the `README.md` as well!

**Comment by trevorb1** [comment]
Also, with the update to `otoole 1.0.4`, there was an issue with default values no longer being written out which was messing up a few of the scripts as sample numpy arrays and result numpy arrays were not matching size. I know otoole has a `--write_defaults` flag, but since this was just for result variables, I hardcoded in default values of `0`. This saves us having to write out super big csv files that are filled with zeros. 

These are the changes that are seen in the `extract_results.py` and `create_heatmap.py` (and associated test) files.

**Comment by trevorb1** [comment]
The `Tutorial.ipynb` file has been updated to work with the changes! There is no example of the scaling, but I think that is outside the current scope of the tutorial notebook. If anyone wants to test it out, that would be great! But not really necessary :)

**Comment by trevorb1** [comment]
I also added a `resources/templates/` folder which has template configuration files for users! Hopefully this will help simplify the process of setting everything up. 

Just thinking about this again though, it may make more sense to put these template files (`parameters.csv`, `results.csv`, `scenarios.csv`, and `config.yaml`) in the `config/` folder to start?

**Comment by trevorb1** [comment]
Updated to `otoole 1.1.0` to take advantage of the better cplex results processing! :)

File: README.md [hunk 1]
```diff
@@ -174,6 +177,48 @@ This repository follows the snakemake guidelines for reproducibility:
     ├── results
     └── resources
 
+## Recreate Paper Results
+
+To recreate the results shown in the accompanying [sensitivity analysis paper](https://doi.org/10.12688/openreseurope.15461.1), follow the steps below. 
+
+1. Pull all data associated with the paper using snakemake 
+```
+snakemake setup_paper_data -c --config skip_checks
```
Comment (willu47) [inline README.md]: This command does not work.  `-c` is related to a cluster.  Should it be `--configfile skip_checks` (but then the `skip_checks` file does not exist)?

File: examples/1/config.yaml
Comment (willu47) [inline examples/1/config.yaml:1]: This file seems to be the otoole.yaml rather than the snakemake configuration file.

File: README.md (hunk 1 above)
Comment (trevorb1) [reply README.md]: I always use `-c` for executing workflows; never knew it was specific for clusters. I guess we should change that to `-j` or `-j6` or something?

**Comment by willu47** [comment]
> Updated to `otoole 1.1.0` to take advantage of the better cplex results processing! :)

Did you update the conda `env/*.yaml` files with pins for the latest otoole?

File: README.md (hunk 1 above)
Comment (trevorb1) [reply README.md]: Changed command to `j6`

File: examples/1/config.yaml
Comment (trevorb1) [reply examples/1/config.yaml:1]: Everything in the `examples/` directory is for the notebook tutorial. I believe example 1 is the demonstration of local SA, therefore the workflow is not run. So, the `config.yaml` file being an otoole configuration file is correct! However, you bring up a good point, that for consistency we should change this to `otoole.yaml` to match the other examples. I will do that!

**Comment by trevorb1** [comment]
> > Updated to `otoole 1.1.0` to take advantage of the better cplex results processing! :)
> 
> Did you update the conda `env/*.yaml` files with pins for the latest otoole?

Good catch! Have updated that now!

File: examples/1/config.yaml
Comment (trevorb1) [reply examples/1/config.yaml:1]: Updated! :)

**Comment by trevorb1** [comment]
Are we able to get rid of this installation command and the `requirements.txt` file, since dependencies are handled through conda now? Not sure if this is just an out-of-date instruction, or if it served some other purpose? 

https://github.com/trevorb1/esom_gsa/blob/933f62b84dc1f47c2a7a08620542058e40f8ccda/README.md?plain=1#L43-L47

**Comment by willu47** [comment]
Otoole is required for a function in the main Snakemake file to run. Until that runs, we need the pip install step.

**Review by willu47 — APPROVED** [review]

**Pull request merged**